features/workflow.feature

```diff
@@ -16,125 +16,127 @@ Feature: Workflow
+        Given an existing pipeline on "master" branch with the workflow jobs:
+            | job       | requires          |
+            | SIMPLE    | ~commit           |
+            | STAGING   | ~commit:staging   |
+            | REGEX     | ~commit:/^.*$/    |
+        When a new commit is pushed to "master" branch
+        Then the "SIMPLE" job is triggered
+        And the "REGEX" job is triggered
+        And the "STAGING" job is not triggered
+        And that "REGEX" build uses the same SHA as the "SIMPLE" build
 
     Scenario: Branch filtering (the staging branch is committed)
-        Given an existing pipeline with the workflow:
-            | job   | requires        |
-            | FOO   | ~commit         |
-            | BAR   | ~commit:staging |
-            | BAZ   | ~commit:/^.*$/  |
-        When a new commit is pushed
-        And it is on the "staging" branch
-        Then the "FOO" job is not started
-        And the "BAR" job is started
-        And the "BAZ" job is started
-        And that "BAZ" build uses the same SHA as the "BAR" build
+        Given an existing pipeline on "master" branch with the workflow jobs:
+            | job       | requires          |
+            | SIMPLE    | ~commit           |
+            | STAGING   | ~commit:staging   |
+            | REGEX     | ~commit:/^.*$/    |
+        When a new commit is pushed to "staging" branch
+        Then the "STAGING" job is triggered
+        And the "REGEX" job is triggered
+        And the "SIMPLE" job is not triggered
+        And that "REGEX" build uses the same SHA as the "STAGING" build
 
+    @ignore
     Scenario: Branch filtering (a pull request is opened to the master branch)
-        Given an existing pipeline with the workflow:
-            | job   | requires        |
-            | FOO   | ~pr             |
-            | BAR   | ~pr:staging     |
-            | BAZ   | ~pr:/^.*$/      |
-            | QUX   | ~pr:master      |
-        When a pull request is opened
-        And it is on the "master" branch
-        Then the "FOO" job is started
-        And the "BAR" job is not started
-        And the "BAZ" job is started
-        And the "QUX" job is started
-        And that "BAZ" and "QUX" build uses the same SHA as the "FOO" build
+        Given an existing pipeline on "master" branch with the workflow jobs:
+            | job       | requires      |
+            | SIMPLE    | ~pr           |
+            | STAGING   | ~pr:staging   |
+            | REGEX     | ~pr:/^.*$/    |
+            | MASTER-PR | ~pr:master    |
+        When a pull request is opened to "master" branch
+        Then the "SIMPLE" job is triggered
+        And the "REGEX" job is triggered
+        And the "MASTER-PR" job is triggered
+        And the "STAGINNG" job is not triggered
+        And that "REGEX" build uses the same SHA as the "SIMPLE" build
+        And that "MASTER-PR" build uses the same SHA as the "SIMPLE" build
 
+    @ignore
```

Comment:
will implementations for this and the following ones come as subsequent PRs ?

Reply:
Yes, we will implement this in another PR.

Reply:
@jithin1987 Yes, we will implement this in another PR.